### DISCUSSION TITLE
Private registry in an air-gap environment used by nodes

### DISCUSSION THREAD

**Discussion by bluefish1976** [rancher/rke2#6807] [Q&A] [answered]
Hello.
In an rke2 cluster in an air-gap environment, can be deployed a private registry of images used by the cluster nodes themselves?
Right now I have made this configuration and when I add the file /etc/rancher/rke2/registries.yaml with the private registry configuration the rke2-agent does not start. If I remove the configuration file rke2-agent starts properly.
The configuration file of registries.yaml it's this one:
mirrors:
  "registry.registry.svc.cluster.local:5000":
    endpoint:
      - "http://registry.registry.svc.cluster.local:5000"
configs:
  "registry.registry.svc.cluster.local:5000"
I don't know if rke2-agent needs the registry to be accessible before start but because there isn't nodes in the cluster to run the registry, the registry never runs and nodes never starts...Like a vicious circle.
Thank you very much, regards.

**Comment by brandond** [comment, marked as answer]
https://docs.k3s.io/installation/registry-mirror

This also works for RKE2.

**Comment by bluefish1976** [comment]
Hello Brad, thank you so much for your support i will try it right now.
